rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -98,7 +98,11 @@ class ReplayRSpace[F[_]: Sync, C, P, A, K](
             )
     } yield r
 
-// TODO: refactor this monster
+  private def filterMatching(comm: COMM)(data: Seq[internal.Datum[A]]) =
+    data.zipWithIndex.filter {
+      case (d, _) => comm.matches(d, produceCounter.get(d.source))
```

Comment:
My nitpick about naming applies here as well. You might consider it redundant, but naming a value based on its type reduces the burden on the programmer to determine the type of the value.

Reply:
Fixed in fb0526681397ec4e067344109bb88797ea6c568d